deployment-tinkerer/src/main/java/org/wso2/testgrid/deployment/tinkerer/AppServletContextListener.java

```diff
@@ -142,4 +151,45 @@ private void sendHeartBeat(SessionManager sessionManager, String agentId) {
             executorService.submit(hbTask);
         }
     }
+
+    /**
+     * Inner class to maintain the message queue. This will persist message queue if it overflow.
+     * Abort execution if it idle for a given time.
+     */
+    static final class MessageQueueScheduler extends TimerTask {
+        /**
+         * Regular inspection cycle of maintaining message queue.
+         */
+        @Override
+        public void run() {
+            for (Map.Entry<String, OperationMessage> operationQueueEntry : SessionManager.getOperationQueueMap()
+                    .entrySet()) {
+                long currentTime = Calendar.getInstance().getTimeInMillis();
+                OperationMessage operationMessage = operationQueueEntry.getValue();
+                final SessionManager sessionManager = SessionManager.getInstance();
+                String deleteOperationId = operationMessage.getOperationId();
+                // If message exist for more longer without consume after the abort operation then remove it from
+                // the message queue
+                long consumeTimeout = Constants.MAX_LAST_CONSUME_TIMEOUT + Constants.MESSAGE_QUEUE_INTERVAL;
+                long updateTimeout = Constants.MAX_LAST_UPDATED_TIMEOUT + Constants.MESSAGE_QUEUE_INTERVAL;
+                if ((operationMessage.getLastConsumedTime() + consumeTimeout < currentTime ||
+                        operationMessage.getLastUpdatedTime() + updateTimeout < currentTime)) {
+                    logger.warn("Operation time out for operation " + deleteOperationId + " " +
+                            operationMessage.getCode() + " deleting message queue");
+                    sessionManager.removeOperationQueueMessages(deleteOperationId);
+                    break;
+                }
+                // Abort operation execution if agent idle or test executor not retrieving back for a given timeout
+                if ((operationMessage.getLastConsumedTime() + Constants.MAX_LAST_CONSUME_TIMEOUT < currentTime ||
+                        operationMessage.getLastUpdatedTime() + Constants.MAX_LAST_UPDATED_TIMEOUT < currentTime) &&
+                        operationMessage.getCode().equals(OperationRequest.OperationCode.SHELL)) {
```

Comment:
Please use Constant.equals(variable) convention to avoid possible null pointers.